### PR TITLE
Add MediaPlayerSupportedFormat

### PR DESCRIPTION
**Pull request by synesthesiam**
Extends the media player info with a list of supported audio formats. This is used in Home Assistant to swap out media URLs going to the ESPHome device with proxy URLs that contain the converted audio.

Also adds supported media formats to the Nabu media player.

Depends on the following PRs:
* https://github.com/home-assistant/core/pull/123254
* https://github.com/esphome/aioesphomeapi/pull/925

### CODE REVIEW

File: esphome/components/media_player/media_player.h [hunk 1]
```diff
@@ -51,6 +51,22 @@ struct MediaFile {
...
+
+struct MediaPlayerSupportedFormat {
+  std::string format;
+  uint32_t sample_rate;
+  uint32_t num_channels;
+  MediaPlayerFormatPurpose purpose;
+};
+
+
+>>>>>>> Stashed changes
```
Comment (nielsnl68) [inline esphome/components/media_player/media_player.h]:
```suggestion
enum class MediaPlayerFormatPurpose : uint8_t {
  DEFAULT = 0,
  ANNOUNCEMENT,
};

struct MediaPlayerSupportedFormat {
  std::string format;
  uint32_t sample_rate;
  uint32_t num_channels;
  MediaPlayerFormatPurpose purpose;
};
```

**Comment by synesthesiam** [comment]
Thanks @nielsnl68

**Comment by jesserockz** [comment]
I think any `api` component changes should be PR'd into the main ESPHome repo instead of this repo as external otherwise we have no idea what is changed here or there. (It's not good to override such core/important components with `external_components`)

**Comment by synesthesiam** [comment]
Closed in favor of: https://github.com/esphome/esphome/pull/7318

**Pull request closed without merging**